lib/xs_transport_lwt_unix_client.ml

```diff
@@ -17,19 +17,38 @@
 open Lwt
 
 (* Individual connections *)
-type channel = Lwt_unix.file_descr * Lwt_unix.sockaddr
+type channel = Lwt_unix.file_descr
+
 let create () =
-  let sockaddr = Lwt_unix.ADDR_UNIX(!Xs_transport.xenstored_socket) in
-  let fd = Lwt_unix.socket Lwt_unix.PF_UNIX Lwt_unix.SOCK_STREAM 0 in
-  lwt () = Lwt_unix.connect fd sockaddr in
-  return (fd, sockaddr)
-let destroy (fd, _) = Lwt_unix.close fd
-let read (fd, _) = Lwt_unix.read fd
-let write (fd, _) bufs ofs len =
-	lwt n = Lwt_unix.write fd bufs ofs len in
-	if n <> len then begin
-		fail End_of_file
-	end else return ()
+  ( match Xs_transport.choose_xenstore_path () with
+    | None ->
+      Printf.fprintf stderr "Failed to find xenstore socket. I tried the following:\n";
+      List.iter (fun x -> Printf.fprintf stderr "  %s\n" x) (Xs_transport.get_xenstore_paths ());
+      Printf.fprintf stderr "\nOn linux you might not have xenfs mounted:\n";
+      Printf.fprintf stderr "   sudo mount -t xenfs xenfs /proc/xen\n";
+      Printf.fprintf stderr "Or perhaps you just need to set the XENSTORED_PATH environment variable.\n";
+      fail Xs_transport.Could_not_find_xenstore
+    | Some x -> return x ) >>= fun path ->
+  Lwt_unix.stat path >>= fun stats ->
+  match stats.Lwt_unix.st_kind with
+  | Lwt_unix.S_SOCK ->
+    let sockaddr = Lwt_unix.ADDR_UNIX(path) in
+    let fd = Lwt_unix.socket Lwt_unix.PF_UNIX Lwt_unix.SOCK_STREAM 0 in
+    Lwt_unix.connect fd sockaddr >>= fun () ->
```

Comment:
I wonder why `xs_transport_unix_client.ml` has a try-with block here (line 35) to close the fd, while this lwt module doesn't, is this specific to lwt?